src/lib.rs

```diff
@@ -270,16 +279,47 @@ impl Server {
                 }
 
                 Some(ctxt.build())
-            }
-            #[cfg(not(feature = "ssl"))]
-            Some(_) => {
-                return Err(
-                    "Building a server with SSL requires enabling the `ssl` feature \
-                                   in tiny-http"
-                        .to_owned()
-                        .into(),
-                )
-            }
+            },
+            #[cfg(feature = "ssl-rustls")]
+            Some(mut config) => {
+                //let mut tls_conf = rustls::ServerConfig::new(rustls::server::NoClientAuth::new());
+
+                let certificate = rustls::Certificate(
+                    match rustls_pemfile::certs(&mut config.certificate.as_slice())?.first() {
+                        Some(cert) => cert.clone(),
+                        None => return Err("Couldn't extract certificate from config.".into())
+                    }
+                );
+                let private_key = rustls::PrivateKey ({
+                    let private_key = config.private_key.clone();
+                    let pkcs8_keys = rustls_pemfile::pkcs8_private_keys(&mut private_key.as_slice())
+                        .expect("file contains invalid pkcs8 private key (encrypted keys not supported)");
+                    // prefer to load pkcs8 keys
+                    if !pkcs8_keys.is_empty() {
+                        pkcs8_keys[0].clone()
+                    } else {
+                        let private_key = config.private_key.clone();
+                        let rsa_keys = rustls_pemfile::rsa_private_keys(&mut private_key.as_slice())
+                            .expect("file contains invalid rsa private key");
+                        rsa_keys[0].clone()
+                    }
+                });
+
+                let tls_conf = rustls::ServerConfig::builder()
+                    .with_safe_defaults()
+                    .with_no_client_auth()
+                    .with_single_cert(vec![certificate], private_key)?;
+
+                // let's wipe the certificate and private key from memory, because we're
+                // better safe than sorry
+                for b in &mut config.certificate { *b = 0; }
+                for b in &mut config.private_key { *b = 0; }
```

Comment:
Sadly the compiler is just going to chuck these two writes in the bin, as the fields are never accessed again.

We be more confident these actually get erased by using [core::ptr::write_volatile](https://doc.rust-lang.org/nightly/core/ptr/fn.write_volatile.html), but rather than try and get that right ourselves let's use the [Zeroize](https://docs.rs/zeroize) helper from RustCrypto.